Gemfile

```diff
@@ -1,6 +1,7 @@
 source 'https://rubygems.org'
 gemspec
 
+gem 'activesupport', '< 7.1'
```

Comment:
This was needed because tests were failing when using `activesupport` `v7.1` – when using the appraisal definitions for `faraday`, the last version of `activesupport` was installed. The error was:

```
NoMethodError:
  undefined method `deprecator' for ActiveSupport:Module
  Did you mean?  deprecate_constant
# ./lib/saddle/method_tree_builder.rb:1:in `require'
# ./lib/saddle/method_tree_builder.rb:1:in `<top (required)>'
# ./lib/saddle.rb:2:in `require'
# ./lib/saddle.rb:2:in `<top (required)>'
# ./spec/spec_helper.rb:2:in `require'
# ./spec/spec_helper.rb:2:in `<top (required)>'
# ./spec/middleware/instrumentation_spec.rb:1:in `require'
# ./spec/middleware/instrumentation_spec.rb:1:in `<top (required)>'
```

It seems like the fix is to require `active_support/deprecator.rb` before requiring `active_support/core_ext/string`.

Reply:
As this is present in the `Gemfile`, this constraint is just for development. People will be able to install the gem with `activesupport` `7.1`. What's more they might not even run into this issue – for example if they are using Rails and requiring `rails` before requiring the gems (which is generally the case).

Having said that, I will open a separate PR for fixing it 🙂

Reply:
https://github.com/mLewisLogic/saddle/pull/22